community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -506,9 +506,13 @@ def sync_maintenance_reservation(lkp: util.Lookup) -> None:
           continue
```

Comment:
We need to check "does nodeset enables reservations" in this case as well.

Reply:
Example implementation:

```py
    for res_name, (node, start_time) in upc_maint_map.items():
      try:
        enabled = lkp.node_nodeset(node).enable_maintenance_reservation
      except Exception:
        enabled = False

      if not enabled:
          if res_name in curr_reservation_map:
            del_reservation.add(res_name)
          continue
      
      if res_name in curr_reservation_map:
        diff = curr_reservation_map[res_name] - start_time
        if abs(diff) <= dt.timedelta(seconds=1):
          continue
        else:
          del_reservation.add(res_name)
          create_reservation_map[res_name] = (node, start_time)
      else:
        create_reservation_map[res_name] = (node, start_time)
```

Reply:
c+v, thanks.